spec/unit/configurer_spec.rb

```diff
@@ -765,4 +796,32 @@ def method_missing(*args)
       @agent.convert_catalog(@oldcatalog, 10)
     end
   end
+
+  describe "when determining whether to pluginsync" do
+    it "should default to Puppet[:pluginsync] when explicitly set by the commandline" do
+      Puppet.settings[:pluginsync] = false
+      Puppet.settings.expects(:set_by_cli?).returns(true)
+
+      expect(described_class.should_pluginsync?).to eq(false)
```

Comment:
Nit, with predicate methods you can do `expect(described_class).to be_should_pluginsync` which can provide a more useful error message if the test fails (instead of "true doesn't equal false"). I'm also fine if you want to leave it as is.